flutter/lib/ui/history/result_list_screen.dart

```diff
@@ -50,27 +56,44 @@ class _ResultListScreenState extends State<ResultListScreen> {
       body: ListView.separated(
         controller: ScrollController(),
         padding: const EdgeInsets.only(top: 20),
-        itemCount: itemList.length,
+        itemCount: itemsList.length,
         separatorBuilder: (context, index) => const Divider(),
         itemBuilder: (context, index) {
-          final item = itemList[index];
-          return ListTile(
-            title: Text(
-              helper.formatDate(item.meta.creationDate.toLocal()),
-              style: const TextStyle(fontWeight: FontWeight.bold),
-            ),
-            subtitle: Text(item.meta.uuid),
-            onTap: () {
-              Navigator.push(
-                context,
-                MaterialPageRoute(
-                  builder: (context) => DetailsScreen(result: item),
-                ),
-              ).then((value) => setState(() {}));
-            },
-          );
+          final item = itemsList[index];
+          return item.build(context);
         },
       ),
     );
   }
+
+  List<ListItem> _listItems(List<ResultItem> resultItems) {
+    return resultItems.map((resultItem) {
+      if (resultItem.item is BenchmarkExportResult) {
+        return _benchmarkListItem(resultItem.item as BenchmarkExportResult);
+      }
+      return _resultListItem(resultItem.item as ExtendedResult);
+    }).toList();
+  }
```

Comment:
The UI for the list of past results should remain consistent in terms of its level of detail, regardless of whether a sort option is applied.

Reply:
Now in the result list we show only BenchmarkExportResult